autoload/go/lint.vim

```diff
@@ -3,28 +3,24 @@ let s:cpo_save = &cpo
 set cpo&vim
 
 function! go#lint#Gometa(bang, autosave, ...) abort
+  let l:metalinter = go#config#Metalinter()
+
   if a:0 == 0
-    let goargs = [expand('%:p:h')]
+    if l:metalinter == "gometalinter"
+      let goargs = [expand('%:p:h')]
+    else
+      let goargs = [expand('%:p')]
+    endif
   else
     let goargs = a:000
   endif
```

Comment:
This block changes things somewhat. When `a:autosave` is not set, then both gometalinter and golangci-lint should be passed the directory. When `a:autosave` is set then `golangci-lint` should be passed the file to check.

Reply:
To be clear, `gometalinter` also needs to be passed the file to check when `a:autosave` is set; that already happens for `gometalinter` a little further down the file. Perhaps this block could be restored to its previous glory, and the section below that ensure that `gometalinter` is given the `--include` option when `a:autosave` is set could be extended to append the file path to `goargs` when `golangci-lint` is used.

Reply:
Cool, that makes sense to me. I did this https://github.com/fatih/vim-go/pull/2182/commits/77243e3c68c32696a62146d06e8f106567f4b3cd instead, do you think it could just be `let goargs = [expand('%p')]`? What all do we expect goargs could be when autosave is enabled?